sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -48,12 +48,12 @@ public static HttpPipeline Build(ClientOptions options, HttpPipelinePolicy[] per
 
             bool isDistributedTracingEnabled = options.Diagnostics.IsDistributedTracingEnabled;
 
+            policies.Add(ClientRequestIdPolicy.Shared);
+
             policies.AddRange(perCallPolicies);
 
             policies.AddRange(options.PerCallPolicies);
 
-            policies.Add(ClientRequestIdPolicy.Shared);
-
             DiagnosticsOptions diagnostics = options.Diagnostics;
```

Comment:
- So there's no user visible change for the default policies because we always initialized the old request ID once per pipeline invocation.
- Moving it higher up the pipeline also shouldn't break anyone who was working around this today by placing their own policy in `PerRetryPolicies` because it will still overwrite.
- The only thing I can imagine breaking is someone's custom `PerCallPolicies` setting `x-ms-client-request-id` will now be respected.  I think that's a safe risk.

Are you okay with this default pipeline policy reordering @KrzysztofCwalina?